### PR TITLE
Bump pyarrow requirement

### LINKED ISSUE #1049: Update pyarrow requirement to support newer versions
## Description
The current pyarrow requirements don't allow pyarrow > 4.0.0, which is old (current version = 6.0.0).

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
## Description
Closes #1049. I'm assuming that if this passes all tests then it's good to merge, but please do say if there's some reason we shouldn't do this.

The range we have for pyarrow now looks very broad, but [their release history](https://pypi.org/project/pyarrow/#history) shows it covers less than 2 years. Just say if we should make this narrower though - happy to do so. 

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by lorenabalan — APPROVED** [review]
Lol yeah it is very broad now - we can maybe bump the minimum to a stable version 1.0.0. I don't have strong feelings either way.

File: RELEASE.md [hunk 1]
```diff
@@ -18,7 +18,7 @@
 * Added support for arbitrary backends (via importable module paths) that satisfy the `pickle` interface to `PickleDataSet`.
 * Added support for `sum` syntax for connecting pipeline objects.
 * Upgraded `pip-tools`, which is used by `kedro build-reqs`, to 6.4. This `pip-tools` version requires `pip>=21.2` while [adding support for `pip>=21.3`](https://github.com/jazzband/pip-tools/pull/1501). To upgrade `pip`, please refer to [their documentation](https://pip.pypa.io/en/stable/installing/#upgrading-pip).
-* Relaxed the bounds on the `plotly` requirement for `plotly.PlotlyDataSet`.
+* Relaxed the bounds on the `plotly` requirement for `plotly.PlotlyDataSet` and the `pyarrow` requirement for `pandas.ParquetDataSet`.
```
Comment (AntonyMilneQB) [inline RELEASE.md:21]:
```suggestion
* Relaxed the bounds on the `plotly` requirement for `plotly.PlotlyDataSet` and the `pyarrow` requirement for `pandas.ParquetDataSet`.
```

File: setup.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def _collect_requirements(requires):
     "pandas.GBQTableDataSet": [PANDAS, "pandas-gbq>=0.12.0, <1.0"],
     "pandas.HDFDataSet": [PANDAS, "tables~=3.6"],
     "pandas.JSONDataSet": [PANDAS],
-    "pandas.ParquetDataSet": [PANDAS, "pyarrow>=0.12.0, <4.0"],
+    "pandas.ParquetDataSet": [PANDAS, "pyarrow>=0.12.0, <7.0"],
```
Comment (AntonyMilneQB) [inline setup.py:81]:
```suggestion
    "pandas.ParquetDataSet": [PANDAS, "pyarrow>=1.0, <7.0"],
```

File: test_requirements.txt [hunk 3]
```diff
@@ -30,7 +30,7 @@ Pillow~=8.0
 plotly>=4.8.0, <6.0
 pre-commit~=1.17
 psutil==5.6.7
-pyarrow>=0.12.0, <4.0
+pyarrow>=0.12.0, <7.0
```
Comment (AntonyMilneQB) [inline test_requirements.txt:33]:
```suggestion
pyarrow>=1.0, <7.0
```

**Review by yetudada — APPROVED** [review]
👍

**Review by limdauto — APPROVED** [review]

**Review by SajidAlamQB — APPROVED** [review]

**Pull request merged**